src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1455,7 +1455,9 @@ static void no_shared_spaces(const char* message) {
     vm_exit_during_initialization("Unable to use shared archive", message);
   } else {
     log_info(cds)("Unable to use shared archive: %s", message);
+#if INCLUDE_CDS
     UseSharedSpaces = false;
+#endif
```

Comment:
This doesn't make sense - the entire `no_shared_spaces` method is only meaningful on a JVM with CDS. Otherwise `-Xshare:on` should immediately be rejected. ??

Reply:
Hi David, unfortunately this has to stay for now, otherwise we would try to set a const in the CDS-disabled build, this does not compile.
Maybe it would make sense to guard the whole function `no_shared_spaces`  and adjust the calls/usages but I would prefer a separate issue for this.

Reply:
AFAICS all calls to `no_shared_spaces` are already with INCLUDE_CDS regions so the function itself can also be INCLUDE_CDS only.